plugins/parse-settings/index.html

```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+
+	<meta charset="utf-8" />
+	<link rel="shortcut icon" href="favicon.png" />
+	<title>Parse Settings ▲ Prism plugins</title>
+	<base href="../.." />
+	<link rel="stylesheet" href="style.css" />
+	<link rel="stylesheet" href="themes/prism.css" data-noprefix />
+	<script src="prefixfree.min.js"></script>
+
+	<script>var _gaq = [['_setAccount', 'UA-33746269-1'], ['_trackPageview']];</script>
+	<script src="http://www.google-analytics.com/ga.js" async></script>
+</head>
+<body>
+
+<header>
+	<div class="intro" data-src="templates/header-plugins.html" data-type="text/html"></div>
+
+	<h2>Parse Settings</h2>
+	<p>Automatically parses settings for other plugins</p>
+</header>
+
+<section class="language-markup">
+	<h1>How to use</h1>
+
+	<p>This plugin is not intended to be used on its own.
+	It is a support plugin for other plugins and only included as a dependency.</p>
+
+	<p>All ancestors of the current element are automatically scanned for
+	<code>data-*</code>-attributes and classes. These attributes and classes are then
+	parsed into key-value pairs and inserted into the <code>env.settings</code>
+	object. An element can inherit the <code>data-*</code>-attributes from its ancestors.</p>
+
+	<p>To avoid accidental matches the element containing the attributes has to be marked for
+	use by Prism. The mark can either be the boolean attribute <code>data-prism</code> or
+	a Prism language class (<code>&lt;pre class="language-javascript"></code>).</p>
```

Comment:
Maybe my description is a bit unclear. The `pre` and `code` tags do not have to be marked:

```html
<pre data-option1="something"><code data-option2="something">
 // Code
</code></pre>
```

Any tag with a prism language class does not have to be marked:

```html
<div class="language-javascript no-setting1 setting2" data-left-trim="false">
<!-- stuff -->
<pre><code>
// code block 1
</code></pre>

<pre><code>
// code block 2
</code></pre>

</div>
```

Only if you want settings on some unrelated tag you have to mark it:

```html
<div class="language-javascript">
<!-- stuff -->
<div data-prism data-left-trim="false">
    <pre><code> // code block 1</code></pre>
</div>

<pre><code>
// code block 2
</code></pre>

</div>
```